ambari-server/src/main/java/org/apache/ambari/server/controller/AmbariManagementControllerImpl.java

```diff
@@ -2095,6 +2099,40 @@ private synchronized RequestStatusResponse updateCluster(ClusterRequest request,
     }
   }
 
+  /**
+   * validate cluster name character and length requirements and throw IllegalArgumentException if not valid.
+   * <p>
+   * Character Requirements
+   * <p>
+   * A through Z
+   * a through z
+   * 0 through 9
+   * _ (underscore)
+   * - (dash)
+   * Length Requirements
+   * <p>
+   * Minimum: 1 character
+   * Maximum: 100 characters
+   * @see AmbariManagementControllerImpl#CLUSTER_NAME_VALIDATION_REGEXP
+   *
+   * @param clusterName name to validate
+   * @throws IllegalArgumentException if validation result
+   */
+  public static void validateClusterName(String clusterName) {
+    if (clusterName == null) {
+      throw new IllegalArgumentException("Invalid arguments, cluster name should not be null");
+    }
+    if (clusterName.isEmpty()) {
+      throw new IllegalArgumentException("Invalid arguments, cluster name should not be empty");
+    }
+    Pattern clusterNamePtrn = Pattern.compile(CLUSTER_NAME_VALIDATION_REGEXP);
```

Comment:
Rather than recompile this, wouldn't it be better to compile this once into a static final property?

Reply:
+1